### PR TITLE
merge 1-15-99 branch into master

### PR DESCRIPTION
**Pull request by tdhock**
hi @MichaelChirico @jangorecki I believe now we can merge the dev branch 1-15-99 into master, right?

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Not quite, we need to `rebase` first, not merge. I've just `--force` pushed the result of rebase, PTAL and make sure the history looks correct.

Crucially, we should _not_ squash commits during this merge so that each commit (which was squashed from a PR into the `1-15-99` branch) survives to `master`.

File: .github/workflows/R-CMD-check.yaml
Comment (MichaelChirico) [inline .github/workflows/R-CMD-check.yaml:7]: once the current run (https://github.com/Rdatatable/data.table/actions/runs/7938397022, https://github.com/Rdatatable/data.table/actions/runs/7938397023) passes, let's remove this carve-out for `1-15-99` from CI as well.

File: NAMESPACE
Comment (MichaelChirico) [inline NAMESPACE:54]: @jangorecki should we roll back #5889 before merging?

File: .dev/CRAN_Release.cmd
Comment (MichaelChirico) [inline .dev/CRAN_Release.cmd:198]: this commit is somewhat strange to survive to `master`, but the changes here still look useful. @TysonStanley shall we add this commit (perhaps with a better commit message) directly to `master` instead of merging along with `1-15-99`?

**Comment by MichaelChirico** [comment]
Hmm, it appears `rebase` messes around with authorship? Now I show up as author of most of the commits :\

Is there a better way to do this, or do I just need to manually edit the commit metadata to restore the "correct" author for each commit & `--force` push again?

File: .github/workflows/R-CMD-check.yaml
Comment (MichaelChirico) [reply .github/workflows/R-CMD-check.yaml]: `drop`'d the relevant commit & force-pushed again

File: .dev/CRAN_Release.cmd
Comment (MichaelChirico) [reply .dev/CRAN_Release.cmd:198]: #5938

File: NAMESPACE
Comment (jangorecki) [reply NAMESPACE:54]: Probably better

Comment (MichaelChirico) [reply NAMESPACE:54]: OK, I hope it doesn't do anything too catastrophic with all the `--force` pushes going around. Since the diffs are still visible in the #5889 web UI, at worst we can reconstitute the changes manually.

Comment (MichaelChirico) [reply NAMESPACE:54]: NB: had to remove the #5910 changes to src/froll.c. Will need to reconstitute those.

**Comment by MichaelChirico** [comment]
OK, current state LGTM. I'll leave it to someone else to merge (again, **not** squash!)

**Review by MichaelChirico — APPROVED** [review]

**Comment by tdhock** [comment]
not sure what is the right way to do that... 
"Create a merge commit" or "Rebase and merge" ? What is the difference?

**Comment by MichaelChirico** [comment]
Not 100% sure, but I went with "Rebase and merge". Seems to have worked -- I believe the former would have added a new commit like "merge branch 1-15-99 to master", which I don't think we want in this case.

**Pull request merged**
